src/pydantic_zarr/core.py

```diff
@@ -16,6 +16,18 @@
 AccessMode: TypeAlias = Literal["w", "w+", "r", "a"]
 
 
+def tuplify_json(obj: object) -> object:
```

Comment:
this function ensures that when we define sequences of ints (like shape and chunk metadata) as tuples, we can deserialize zarr metadata from JSON and get tuples there too, so we can avoid spurious errors due to the in-memory side using a tuple and the JSON side using a list to represent the same data

Reply:
I presume we can't just use lists on the in-memory side?

Reply:
i would rather not. lists are mutable, which complicates things like e.g., inheritance for typeddicts, which would be rather useful for us. And since ArraySpec and GroupSpec should be immutable we don't lose anything by making their fields immutable.